### PR TITLE
Add classes supporting requirements with distinct options

### PR DESCRIPTION
**Pull request by dimits-exe**

### CODE REVIEW

**Review by AAAlex-123 — CHANGES_REQUESTED** [review]
Can't compile.

List Requirements can be used:
 - File Menu: open, import (Find all appropriate files (ending in .scad) from directory. Languages.java has some code that may help).
 - Whenever an id of an already created component is needed (Find all Components from the Editor. Optionally filter them to display only the ones that are applicable for the specific operation. Not necessary, since checks already exist.).
 - Language selection (Tweak the pop-up dialog).

For every change:
 - Remove unused code, possibly from other classes
 - Use existing externalized Strings
 - Test everything to make sure that nothing broke and all checks and error messages work as before.

File: src/localisation/Languages.java [hunk 1]
```diff
@@ -128,9 +130,11 @@ public static boolean editAndWriteToFile(Frame frame) throws IOException {
...
 		}
 
-		Locale chosen  = (Locale) JOptionPane.showInputDialog(frame,
-		        Languages.getString("Languages.3"), Languages.getString("Languages.4"),                            //$NON-NLS-1$ //$NON-NLS-2$
-		        JOptionPane.PLAIN_MESSAGE, null, locales.toArray(), locales.get(0));
+		Requirements reqWrapper = new Requirements();
+		reqWrapper.add("languages", locales);
+		reqWrapper.fulfillWithDialog(frame, "languages");
+		Locale chosen = (Locale) reqWrapper.getValue("languages");
+
```
Comment (AAAlex-123) [inline src/localisation/Languages.java:137]: Use the same Languages.getString() parameters as the previous version

File: src/requirement/requirements/Requirements.java [hunk 2]
```diff
@@ -88,6 +89,17 @@ public void add(String key, StringType type) {
...
 	}
+	
+	/**
+	 * Adds a {@link ComponentRequirement} with a {@code key} and {@code components}.
+	 *
+	 * @param key  the key
+	 * @param components the suitable components
+	 *
+	 */
+	public void addComponentReq(String key, List<components.Component> components) {
```
Comment (AAAlex-123) [inline src/requirement/requirements/Requirements.java:100]: rename to add(String key, List<Component> components) 🔪

File: src/localisation/Languages.java (hunk 1 above)
Comment (dimits-exe) [reply src/localisation/Languages.java:137]: οχι 😎

File: src/application/editor/Actions.java [hunk 3]
```diff
@@ -72,6 +76,13 @@ public void execute() {
 
 	/** Action for deleting a {@code Component} */
 	DELETE(EditorStrings.COMMAND) {
+		
+		@Override
+		public Actions specifyWithDialog(Editor editor) {
+			((ComponentRequirement) reqs.get(EditorStrings.COMMAND)).setComponentOptions(editor.getComponents_());
+			return super.specifyWithDialog(editor);
+		}
```
Comment (AAAlex-123) [inline src/application/editor/Actions.java:84]: Wrong place, already implemented in DeleteCommand#fillRequirements(Frame, Editor).

**Review by AAAlex-123 — APPROVED** [review]
Looks good 👍

**Pull request merged**